workflow-ui/compose/src/main/java/com/squareup/workflow1/ui/compose/RenderAsState.kt

```diff
@@ -185,7 +204,13 @@ private class WorkflowRuntimeState<PropsT, OutputT : Any, RenderingT>(
         renderingState.value = it.rendering
         snapshotState.value = it.snapshot
       }
-      .launchIn(workflowScope)
+      // We collect the renderings in the workflowScope to participate in structured concurrency,
+      // however we don't need to use its dispatcher – this coroutine is simply setting snapshot
+      // state values, which is threadsafe.
+      // Also, if the scope uses a non-immediate dispatcher, the initial states won't get set until
+      // the dispatcher dispatchers the collection coroutine, but our contract requires them to be
+      // set by the time this function returns.
+      .launchIn(workflowScope + Dispatchers.Unconfined)
```

Comment:
These seem to be important changes buried in this "migrate samples" PR. Could you please add some notation in the commit history?

Reply:
Oops, I forgot to retarget this pr. It's based on this branch: https://github.com/square/workflow-kotlin/pull/503